prov/gni/src/gnix_ep.c

```diff
@@ -1519,12 +1612,14 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	ep_priv->progress_fn = NULL;
 	ep_priv->rx_progress_fn = NULL;
 
-	ret = gnix_nic_alloc(domain_priv, attr, &ep_priv->nic);
-	if (ret != FI_SUCCESS) {
-		GNIX_WARN(FI_LOG_EP_CTRL,
-			    "_gnix_nic_alloc call returned %d\n",
-			     ret);
-		goto err;
+	if (ep_priv->nic == NULL) {
```

Comment:
I think I get it, only the first EP on the domain creates a CM NIC and uses that same GNI NIC for itself (which is where resources are saved).  We still need to create a new NIC for additional EPs on the NIC, though?